src/metrics-gatherer.ts

```diff
@@ -112,6 +112,21 @@ export class MetricsGatherer {
 		}
 	}
 
+	// increment/decrement a gauge during the runtime of a function
+	public async run(name: string, func: () => any, labels: LabelSet = {}) {
+		this.inc(name, 1, labels);
+		await func();
+		this.inc(name, -1, labels);
```

Comment:
Matching the other function this should be
```suggestion
		try {
			await func();
		} finally {
			this.inc(name, -1, labels);
		}
```
and make sure we don't drop decrements